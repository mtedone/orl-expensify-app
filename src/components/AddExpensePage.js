import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { addExpense } from '../actions/expenses';

export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.addExpense(expense);
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <h1>Add Expense</h1>
        <ExpenseForm onSubmit={this.onSubmit} />
      </div>
    );
  }
}

// Refactors dispatch(addExpense) here so the component is more testable
const mapDispatchToProps = (dispatch) => ({
  addExpense: (expense) => dispatch(addExpense(expense)),
});

// First argument is mapStateToProps, the second is mapDispatchToProps
export default connect(undefined, mapDispatchToProps)(AddExpensePage);
