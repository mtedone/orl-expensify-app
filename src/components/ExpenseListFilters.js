import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByDate, sortByAmount } from '../actions/filters';

const ExpenseListFilters = (props) => (
  <div>
    <input
      type="text"
      value={props.filters.text}
      onChange={(e) => {
        // dispatch is available from components
        props.dispatch(setTextFilter(e.target.value));
      }}
    />
    <select
      value={props.filters.sortBy}
      onChange={(e) => {
        const val = e.target.value;
        if (val === 'date') {
          props.dispatch(sortByDate());
        } else if (val === 'amount') {
          props.dispatch(sortByAmount());
        }
      }}
    >
      <option value="date">Date</option>
      <option value="amount">Amount</option>
    </select>
  </div>
);

const mapStateToProps = (state) => {
  return {
    filters: state.filters,
  };
};

export default connect(mapStateToProps)(ExpenseListFilters);
