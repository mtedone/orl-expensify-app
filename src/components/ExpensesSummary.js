import React from 'react';
import { connect } from 'react-redux';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = (props) => (
  <div>
    {props.expensesCount === 1 && (
      <p>
        Viewing 1 expense totalling
        {' ' + numeral(props.expensesTotal / 100).format('$0,0.00')}
      </p>
    )}
    {props.expensesCount > 1 && (
      <p>
        Viewing {props.expensesCount} expenses totalling{' '}
        {' ' + numeral(props.expensesTotal / 100).format('$0,0.00')}
      </p>
    )}
  </div>
);

const mapStateToProps = (state, props) => ({
  expensesCount: state.expenses.length,
  expensesTotal: selectExpensesTotal(state.expenses),
});

export default connect(mapStateToProps)(ExpensesSummary);
