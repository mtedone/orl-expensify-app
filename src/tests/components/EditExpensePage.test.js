import React from 'react';
import { shallow } from 'enzyme';
import { EditExpensePage } from '../../components/EditExpensePage';
import expenses from '../fixtures/expenses';

let wrapper, editExpenseSpy, removeExpenseSpy, historySpy;

beforeEach(() => {
  editExpenseSpy = jest.fn();
  removeExpenseSpy = jest.fn();
  historySpy = {
    push: jest.fn(),
  };
  wrapper = shallow(
    <EditExpensePage
      editExpense={editExpenseSpy}
      removeExpense={removeExpenseSpy}
      history={historySpy}
    />
  );
});

test('should render EditExpensePage', () => {
  expect(wrapper.debug()).toMatchSnapshot();
});

test('should handle editExpense', () => {
  wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1]);

  expect(historySpy.push).toHaveBeenLastCalledWith('/');
  expect(editExpenseSpy).toHaveBeenLastCalledWith(expenses[1]);
});
