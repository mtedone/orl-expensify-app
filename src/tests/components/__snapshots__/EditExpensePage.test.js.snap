// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpensePage 1`] = `
"<div>
  <ExpenseForm expense={[undefined]} onSubmit={[Function]} />
  <button onClick={[Function]}>
    Remove
  </button>
</div>"
`;
